torch/_subclasses/fake_tensor.py

```diff
@@ -331,6 +334,19 @@ def __init__(self, fake_mode, elem, device: Union[torch.device, str]):
         if device.type == "cuda" and device.index is None:
             device = torch.device(f"cuda:{torch.cuda.current_device()}")
         assert device.type != "meta"
+
+        # Note: [Fake Tensor Dispatch Keys]
+        # In order to model the behavior of device-specific autocast
+        # and autograd logic, we update the dispatch keys of FakeTensors
+        # to reflect their fake device. This includes the BackendComponent
+        # (DispatchKey::Meta -> DispatchKey::CUDA), and also the BackendComponent
+        # related Autocast and Autograd keys. __torch__dispatch__ sits below
+        # Autocast and Autograd, and is only invoked when we are at the
+        # kernel for the BackendComponent. Then, we add Meta to the
+        # thread-local dispatch include set to hit the meta kernel
+        # instead of the kernel of the BackendComponent for the fake device.
+
+        torch._C._change_backend_component_keys(self, device)
```

Comment:
Yeah, my preference is probably to have this be an argument to `_make_subclass`, but if you need to land sooner than later this is OK. The big problem with this API is it lets people transmute backend key mid-Tensor lifetime and that's going to be very confusing.

Reply:
If you don't have access to expensive GPUs you can now `_change_backend_component_keys` on your CPU tensors to get access to all those great kernels 😉